native/core/src/parquet/mod.rs

```diff
@@ -620,12 +619,21 @@ fn get_batch_context<'a>(handle: jlong) -> Result<&'a mut BatchContext, CometErr
     }
 }
 
-/*
-#[inline]
-fn get_batch_reader<'a>(handle: jlong) -> Result<&'a mut ParquetRecordBatchReader, CometError> {
-    Ok(&mut get_batch_context(handle)?.batch_reader.unwrap())
+fn get_file_groups_single_file(
+    path: &Path,
+    file_size: u64,
+    start: i64,
+    length: i64,
+) -> Vec<Vec<PartitionedFile>> {
```

Comment:
just wondering why file groups are a 2d vector?

Reply:
Outer vector is partitions, inner vector is the files belonging to that partition, I believe:

https://github.com/apache/datafusion/blob/f8828abb46d6f9280f4de146896cd59a740e06ed/datafusion/datasource/src/file_scan_config.rs#L145